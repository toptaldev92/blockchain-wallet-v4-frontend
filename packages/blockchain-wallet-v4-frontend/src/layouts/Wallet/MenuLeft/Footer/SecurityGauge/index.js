import React from 'react'
import { compose, bindActionCreators } from 'redux'
import { connect } from 'react-redux'

import { actions } from 'data'
import { getData } from './selectors'
import { SecurityGauge } from 'blockchain-info-components'

class SecurityGaugeContainer extends React.PureComponent {
  render () {
    const { data } = this.props

    return data.cata({
      Success: value => <SecurityGauge score={value} />,
      Failure: message => <div />,
      Loading: () => <div />,
      NotAsked: () => <div />
    })
  }
}

const mapStateToProps = state => ({
  data: getData(state)
})

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(actions.core.settings, dispatch)
})

const enhance = compose(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)

export default enhance(SecurityGaugeContainer)
