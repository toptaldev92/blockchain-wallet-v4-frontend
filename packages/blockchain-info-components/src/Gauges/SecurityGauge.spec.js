import { shallow } from 'enzyme'
import React from 'react'
import toJson from 'enzyme-to-json'

import SecurityGauge from './SecurityGauge'

describe('SecurityGauge component', () => {
  it('default renders correctly', () => {
    const component = shallow(<SecurityGauge score={1} />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
