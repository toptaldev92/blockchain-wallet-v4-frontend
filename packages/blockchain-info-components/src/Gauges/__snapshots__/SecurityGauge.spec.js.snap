// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecurityGauge component default renders correctly 1`] = `
<SecurityGauge__Progress>
  <Styled(Icon)
    name="checkmark-in-circle"
    size="16px"
  />
  <SecurityGauge__Empty />
  <SecurityGauge__Empty />
</SecurityGauge__Progress>
`;
